babel/src/board.js

```diff
@@ -67,15 +67,19 @@ export function iterchunks (orbs, chunkLimitRange = [4, 2]) {
 
 export function findMatches(orbs) {
     let matches = [];
-    let b = orbs;
     let height = orbs.length;
     let width = orbs[0].length;
 
-    _.each(_.range(height - 2), x => {
+    _.each(_.range(height), x => {
         _.each(_.range(width - 2), y => {
-            if (b[x][y] == b[x][y + 1] && b[x][y] == b[x][y + 2]) {
+            if (orbs[x][y] == orbs[x][y + 1] && orbs[x][y] == orbs[x][y + 2]) {
                 matches.push([[x, y], [x, y + 1], [x, y + 2]])
-            } else if (b[x][y] == b[x + 1][y] && b[x][y] == b[x + 2][y]) {
+            };
+        });
+    });
+    _.each(_.range(height - 2), x => {
+        _.each(_.range(width), y => {
+            if (orbs[x][y] == orbs[x + 1][y] && orbs[x][y] == orbs[x + 2][y]) {
```

Comment:
A couple things here:

1. Can you use a technique like I did in iterchunks, and make an inner function that only checks horizontally, and feed it both the orbs, and then the transposed orbs? It'll de-duplicate your for loop logic.

2. Does it make more sense to grab each 3x1 section of the board, and mark them as matches if they're all the same? That's way less code, and `combineMatches` could do the rest. I don't think there's a need to get the next orb, the previous one, the one above, etc., when finding matches. You can just grab a given slice of three, and check that all three are the same.